[clang] Add tests for CWG1800-1804

Endilll opened the pull request:
Covers C++ core issues 1800, 1801, 1802, 1803, 1804.

Endilll committed:
[clang] Add tests for CWG1800-1804

Covers C++ core issues 1800, 1801, 1802, 1803, 1804.

Endilll commented:
This PR also updates status for CWG437, which is a follow-up to #77372.
Nothing wrong with the tooling, just my mistake separating output of make_cxx_dr_status from changes I already had for complete-class context DRs.

cor3ntin commented on clang/test/CXX/drs/dr18xx.cpp:
```diff
@@ -1,16 +1,169 @@
+#if __cplusplus >= 201103L
+// Using a Wikipedia example of surrogate pair:
+// https://en.wikipedia.org/wiki/UTF-16#Examples
+constexpr char16_t a[3] = u"𐐷";
```
This is really just an array of `char16_t` - so `𐐷` would be parsed as 2 char16_t
What the wording is referring to is that any of the following should yield 2 elements 

```cpp
u"\N{DESERET SMALL LETTER YEE}"
u"\u{10437}"
u"\U00010437"
```

cor3ntin replied:
https://eel.is/c++draft/lex.charset#nt:universal-character-name

Endilll replied:
Thank you! Updated the test to use `\U00010437` as the first two forms you provided are too new for older language modes.

cor3ntin commented on clang/test/CXX/drs/dr18xx.cpp:
```diff
@@ -1,16 +1,169 @@
+} // namespace dr1802
+
+namespace dr1803 { // dr1803: 2.9
+#if __cplusplus >= 201103L
```
@erichkeane I think it looks reasonable but I would like a second pair of eyes
https://cplusplus.github.io/CWG/issues/1804.html

Endilll replied:
(Test for 1804 is at line 56)

erichkeane replied:
Yeah, I think the test for 1804 tests everything I can thing of.

Endilll committed:
Update CWG1802 test

Replace a character that has to be encoded as a surrogate pair with a universal character name.

cor3ntin reviewed (approved):
LGTM
